book/src/getting-started/5-cmake-integration.md

```diff
@@ -0,0 +1,153 @@
+- Add `cxx`, `cxx-qt`, as well as `cxx-qt-lib` as dependencies.
+- Add `clang-format` and `cxx-qt-build` as build-dependencies.
+
+In the end, your `Cargo.toml` should look like this:
+```toml,ignore
+[package]
+name = "cxx-qt-getting-started"
+version = "0.1.0"
+edition = "2021"
+
+# Don't touch this, the CMake file expects a static lib with a specific name
+[lib]
+name = "rust"
+crate-type = ["staticlib"]
+
+[dependencies]
+cxx = "1.0"
+cxx-qt = "0.2"
+cxx-qt-lib = "0.2"
+
+[build-dependencies]
+clang-format = "0.1"
+cxx-qt-build = "0.2"
+```
+
+We'll then also need to add a script named `build.rs` next to our `Cargo.toml`:
+```rust,ignore
+{{#include ../../../examples/qml_minimal/build.rs:book_build_rs}}
+```
+This is what generates the C++ code for our `MyObject` class at compile-time.
+It will output the `cxx-qt-gen/include/my_object.h` file we included earlier in `main.cpp`.
+
+Then we can write our `CMakeLists.txt` file:
+
+```cmake,ignore
+cmake_minimum_required(VERSION 3.16)
+
+project(cxx_qt_getting_started)
+set(APP_NAME ${PROJECT_NAME})
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+find_package(QT NAMES Qt6 Qt5 COMPONENTS Core Gui Qml QuickControls2 QuickTest Test REQUIRED)
+find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui Qml QuickControls2 QuickTest Test REQUIRED)
+
+include(CxxQt)
+
+# Generate C++ code from Rust using Cargo in the current folder
+cxx_qt_generate_cpp(GEN_SOURCES)
+
+# Define our sources
+set(
+    CPP_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/main.cpp
+)
+
+set(
+    RESOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/qml.qrc
+)
+
+# Define our executable with our C++ source, generated sources, and QML resource files
+add_executable(${APP_NAME} "${CPP_SOURCES}" "${GEN_SOURCES}" "${RESOURCES}")
+
+# Include generated sources
+cxx_qt_include(${APP_NAME})
+
+# Link to generated rust library
+cxx_qt_link_rustlib(${APP_NAME})
+
+# Link to Qt in the normal way
+target_link_libraries(${APP_NAME}
+    Qt${QT_VERSION_MAJOR}::Core
+    Qt${QT_VERSION_MAJOR}::Gui
+    Qt${QT_VERSION_MAJOR}::Qml
+    Qt${QT_VERSION_MAJOR}::QuickControls2
+)
+```
+
+This looks like a lot, but it is actually just a fairly standard CMake file for building a Qt application.
+
+The difference here are these lines:
+```cmake,ignore
+include(CxxQt)
+
+# Generate C++ code from Rust using Cargo in the current folder
+cxx_qt_generate_cpp(GEN_SOURCES)
+
+# Include generated sources
+cxx_qt_include(${APP_NAME})
+
+# Link to generated rust library
+cxx_qt_link_rustlib(${APP_NAME})
+```
+
+Which will do the code generation and include it into the C++ build.
+
+An important thing to note here is that CMake must be able to resolve the call to `include(CxxQt)`.
+For this to work, you'll want to clone the [CXX-Qt repository](https://github.com/KDAB/cxx-qt/) and add the `CxxQt.cmake` file to the `CMAKE_MODULE_PATH` CMake variable.
```

Comment:
Do we want to note that something like `list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../cmake")` is a way of doing that? (this is what we do in the existing CMakeLists.txt to add it to the module path)/

Reply:
I think this is something for the `concepts` chapter to take care of.
Here I just want to show the "easiest" option.
For something unfamiliar with CMake, I think this should be easy enough.
If we add this info to the dedicated CMake section , we can link to it from here.

Reply:
We do have the `/concepts/cmake.html` section now, so lets add it there...